### PR TITLE
Avoid re-running glock again after a "git pull".

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

**Comment by tamird** [comment]
Isn't this just papering over the issue that glock is needlessly recompiling protoc? and that that operation isn't instant?

**Comment by petermattis** [comment]
Glock needlessly recompiling protoc is a different issue. This is fixing the annoyance that we've already applied the glock changes so there is no need to run glock again the next time `make` is invoked.

**Comment by tamird** [comment]
LGTM

**Comment by bdarnell** [comment]
Bootstrapping is more than glock - it also installs javascript dependencies. We don't have the right dependencies for those in the makefile yet but if we did we wouldn't want to suppress further bootstrapping just because the glockfile changed.

In fact, now that we have the `.bootstrap` rule in the makefile, maybe we should just rely on that and remove the git hooks.

**Comment by petermattis** [comment]
Or maybe the `githooks` should run `make bootstrap` when a change to `GLOCKFILE` is detected.

**Comment by petermattis** [comment]
Any thoughts about having the `githooks` run `make bootstrap`? This should avoid the double execution of `glock` and not have problems with accidentally skipping other dependencies.

**Comment by tamird** [comment]
SGTM

**Comment by bdarnell** [comment]
SGTM. I think the git hooks should simply run `make bootstrap` no matter what changed; let `make` decide whether anything needs to be updated. This way when we add proper dependency tracking for the javascript dependencies we don't need to update two different places.

**Comment by petermattis** [comment]
This is ready for another look.

File: build/devbase/deps.sh [hunk 1]
```diff
@@ -23,18 +23,6 @@ fi
...
-# adding /... to repo paths will match packages that have been downloaded but
-# whose dependencies have not).
-pkgs=$(cat build/devbase/deps)
-
 set -x
 
-# NOTE: "glock sync" downloads the source but does not install libraries
-# into GOPATH/pkg. Install both race and non-race builds here.
-go install ${pkgs}
-go install -race ${pkgs}
```
Comment (petermattis) [inline build/devbase/deps.sh]: Pretty sure this isn't necessary now that we're not building a base docker image for development.

**Comment by tamird** [comment]
LGTM

**Pull request merged**